File: Sorting/1200.Minimum_Absolute_Difference/Minimum_Absolute_Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        min_dif = float("inf")
        ans = []
        arr = sorted(arr)
        for i in range(1, len(arr)):
            curr_dif = arr[i] - arr[i - 1]
            if  curr_dif < min_dif: 
                ans = [[arr[i - 1], arr[i]]]
                min_dif = curr_dif
            elif curr_dif == min_dif:
                ans.append([arr[i - 1], arr[i]])
        return ans

File: Sorting/1200.Minimum_Absolute_Difference/Minimum_Absolute_Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        int min_diff = numeric_limits<int>::max();
        vector<vector<int>> ans;
        sort(arr.begin(), arr.end());
        for(int i = 1; i < arr.size(); i++){
            int curr_diff = arr[i] - arr[i - 1];
            if(curr_diff < min_diff){
                min_diff = curr_diff;
                ans = {{arr[i - 1], arr[i]}};
            }
            else if (curr_diff == min_diff){
                ans.push_back({arr[i - 1], arr[i]});
            }
        }
        return ans;
    }
};
